tests/test_qnode.py

```diff
@@ -1130,3 +1130,43 @@ def circuit(x):
 
         qml.grad(circuit)(x)
         assert len(spy_expand.call_args_list) == 3
+
+    def test_expansion_multiple_qwc_observables(self, mocker):
+        """Test that the QNode correctly expands tapes that return
+        multiple measurements of commuting observables"""
+        dev = qml.device("default.qubit", wires=2)
+        obs = [qml.PauliX(0), qml.PauliX(0) @ qml.PauliY(1)]
+
+        @qml.qnode(dev)
+        def circuit(x, y):
+            qml.RX(x, wires=0)
+            qml.RY(y, wires=1)
+            return [qml.expval(o) for o in obs]

```

Comment:
We did not have any tests on returning multiple expvals before that could detect the unexpected behaviour?